src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyName.cs

```diff
@@ -200,11 +198,8 @@ public AssemblyVersionCompatibility VersionCompatibility
             set => _versionCompatibility = value;
         }
 
-        public StrongNameKeyPair? KeyPair
-        {
-            get => _strongNameKeyPair;
-            set => _strongNameKeyPair = value;
-        }
+        [Obsolete("Strong name signing is not supported on this platform.")]
+        public StrongNameKeyPair? KeyPairr => throw new PlatformNotSupportedException(SR.PlatformNotSupported_StrongNameSigning);
```

Comment:
We can't get rid of the backing field. The native reflection stack requires it to be present. So this property getter and setter may have to remain present. Let's hold of on obsoleting this property for now and revert it back to what it was.

Reply:
We have a few options for how to respond to this.

1. Obsolete the property, but leave its implementation as-is. You can get and set the property. And if you're doing so while targeting .NET 6, you'll receive a warning.
2. Change the implementation of the property getters / setters to no-op or throw. This maybe an unacceptable breaking change, as it looks like there are [many consumers](https://source.dot.net/System.Private.CoreLib/R/64dc4409ac56c451.html) of this API.
3. Do nothing with the property. Do not mark it obsolete. If the backing type `StrongNameKeyPair` is marked obsolete, then callers will see a warning anyway.

Reply:
I am leaning to #1 it is not really used within runtime, and as you mentioned callers will still get the warning from `StrongNameKeyPair` there is no much difference i think. But if you would suggest #3, I will just suppress the warning for the property

Reply:
`#1` seems fine. Consider: After this change, you'll never be able to instantiate a `StrongNameKeyPair` again. So the only thing you'll ever be able to do is set _KeyPair_ to null or retrieve null from the _KeyPair_ property. Totally fine IMO to mark the properties as obsolete since they're completely useless at that point.

I suppose we could also remove the native reflection support for this as well at the same time, but that's up to you.

Buyaa, what do you think about defining the property as `get => null; set => { };`? That would allow you to delete the backing field as you had originally wanted to do, assuming you also delete the following lines from the native reflection stack at the same time.

https://github.com/dotnet/runtime/blob/927b1c54956ddb031a2e1a3eddb94ccc16004c27/src/coreclr/vm/object.h#L1505

https://github.com/dotnet/runtime/blob/927b1c54956ddb031a2e1a3eddb94ccc16004c27/src/coreclr/vm/object.h#L1518

Reply:
> I suppose we could also remove the native reflection support for this as well at the same time, but that's up to you.

From my search seems that is not referenced anywhere, so probably we could remove that

> Buyaa, what do you think about defining the property as get => null; set => { };? That would allow you to delete the backing field as you had originally wanted to do, assuming you also delete the following lines from the native reflection stack at the same time.

The only concern is those references from MSBuild you mentioned in `#2`, i saw them before updating to `throw PNSE`, but thought those are old/stale probably unused/unreachable code as the type itself is not doing anyways, so i was expecting MSBuild would need some cleanup, but my expectation might be wrong, maybe we should talk with somebody from MSBuild?

Reply:
> maybe we should talk with somebody from MSBuild?

A good idea, certainly! :)

Reply:
I am not sure whom to talk to, so i created an [issue](https://github.com/dotnet/msbuild/issues/6343) in MsBuild repo